### PR TITLE
[Feature] Detect maximum cluster and move

### PR DESCRIPTION
**Pull request by MFizz**
Short Description: 
Detects the location where there are the most lured(if set in config) pokestops in a given radius (Uses maximum clique algorithm to find the maximum points and mercator projection to compute the average) and walks there. 
As new stops are being lured and old expire, destiniy will change over time. If nothing changes bot will stay in the middle of reachable (as set in radius - perhaps more than 50m?) lured pokestops. With no lured pokestops in the vicinity, it will search for the biggest non-lured cluster instead.
Also if someone knows/tests the maximum distance for spinning forts, one can set that as default (or even remove the config option)

Now in utils there is find_biggest_cluster(radius, points) which can be used to find any kind of clusters, e.g. lured pokestops, pokemons, etc. if you have their lat, lng

Goes into direction of #1498

Got some example pictures:
cover2
graph

cover1
red has not the right scaled area but determines the mean

### CODE REVIEW

**Comment by douglascamata** [comment]
Lock versions in `requirements.txt`, please.

**Comment by douglascamata** [comment]
Need some tests feedback.

File: pokecli.py [hunk 1]
```diff
@@ -249,6 +249,16 @@ def init_config():
     add_config(
         parser,
         load,
+        short_flag="-r",
```
Comment (TheSavior) [inline pokecli.py]: This should be a task level configuration option. Check out IncubateEggs for an example

**Comment by MFizz** [comment]
Changed accordingly

File: pokemongo_bot/cell_workers/find_biggest_cluster.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+from pokemongo_bot.cell_workers.utils import distance, merc2coord, coord2merc
+from networkx.algorithms.clique import find_cliques
+
+import networkx as nx
+import numpy as np
+
+
+class FindBiggestCluster(object):
```
Comment (TheSavior) [inline pokemongo_bot/cell_workers/find_biggest_cluster.py]: Are you able to combine the FindBiggestCluster and FollowBiggestCluster? Our other navigators are only a single task. Perhaps you can take a look at those for examples.

File: pokemongo_bot/cell_workers/find_biggest_cluster.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    def _process_config(self):
+        self.radius = self.config.get("radius", 50)
+
+    def work(self):
+        forts = self.bot.get_forts()
+        self.dest = find_biggest_cluster(self.radius, forts)
+
+
+def find_biggest_cluster(radius, points):
```
Comment (TheSavior) [inline pokemongo_bot/cell_workers/find_biggest_cluster.py]: This isn't required, we don't have tests for many other things, but would you be willing to write a few unit tests for this? It seems like something that would benefit greatly from having a few test cases proving that it works and that we don't break it in future PRs. We have some tests for some other classes already in the tests folder.

**Comment by TheSavior** [comment]
I'm really quite excited about this. I think others will be as well.

File: pokemongo_bot/cell_workers/find_biggest_cluster.py (hunk 2 above)
Comment (MFizz) [inline pokemongo_bot/cell_workers/find_biggest_cluster.py]: Will do

File: pokemongo_bot/cell_workers/find_biggest_cluster.py (hunk 3 above)
Comment (MFizz) [inline pokemongo_bot/cell_workers/find_biggest_cluster.py]: Tests for what functionality do you have in mind?

File: pokemongo_bot/cell_workers/find_biggest_cluster.py (hunk 2 above)
Comment (MFizz) [inline pokemongo_bot/cell_workers/find_biggest_cluster.py]: On another note. The original idea was to seperate the cluster-finder because it can find clusters of any kind not just forts. So for future functionality it could perhaps find the biggest pokets of pokemons for example. Thats why perhaps it would be a good idea to be seperate, so other tasks could use it? I could put it into utils.

**Comment by MFizz** [comment]
I'll write some tests

**Comment by MFizz** [comment]
Do you want more functionalities tested, or do you want the tests to test more specific?

**Comment by douglascamata** [comment]
@MFizz dunno if it's possible to check if it is lured from far away, but when we're close to it we can. Want me to w8 for u to add more features or can I merge it now?

**Comment by douglascamata** [comment]
Seems like you got trouble in a PR tho.... the diff show some file changes that are out of the scope of your features.

**Comment by MFizz** [comment]
The navigator works as is, but can be extended with a lured version in future PRs if possible.
I made a rebase on the current dev branch - which are the problematic commits/ what do you recommend - seems like I there are new merge conflicts too with the new commits.

**Comment by MFizz** [comment]
Made a new rebase and cleanup

**Comment by douglascamata** [comment]
👎

**Comment by douglascamata** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/PokemonGoF/PokemonGo-Bot/pull-request/1993/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=PokemonGoF/PokemonGo-Bot)

**Comment by 1cu** [comment]
Bot finds a cluster and stays there completely immobile. Is that intended?

**Comment by douglascamata** [comment]
@1cu show your config

**Comment by MFizz** [comment]
> Bot finds a cluster and stays there completely immobile. Is that intended?

Yes, since the normal pokestops don't change thats indeed the behaviour. Originally it was intended to use only lured stops, which would have changed over time. But I don't know how to search for lured stops yet.

**Comment by douglascamata** [comment]
@MFizz maybe https://github.com/PokemonGoF/PokemonGo-Bot/pull/2068 has information about how to detect if a fort has a lure.

**Comment by 1cu** [comment]
`{
    "auth_service": "google",
    "username": "...",
    "password": "...",
    "location": "",
    "gmapkey": "...",
    "tasks": [
      {
        "type": "HandleSoftBan"
      },
      {
        "type": "CollectLevelUpReward"
      },
      {
        "type": "IncubateEggs",
        "config": {
          "longer_eggs_first": true
        }
      },
      {
        "type": "TransferPokemon"
      },
      {
        "type": "EvolveAll",
        "config": {
          "evolve_speed": 20,
          "use_lucky_egg": false
        }
      },
      {
        "type": "RecycleItems"
      },
      {
        "type": "CatchVisiblePokemon"
      },
      {
        "type": "CatchLuredPokemon"
      },
      {
        "type": "SpinFort"
      },
      {
        "type": "FollowCluster",
        "config": {
            "radius": 50
        }
      }
    ],
    "max_steps": 5,
    "forts": {
      "avoid_circles": true,
      "max_circle_size": 50
    },
    "websocket_server": false,
    "walk": 6.16,
    "action_wait_min": 1,
    "action_wait_max": 4,
    "debug": false,
    "test": false,
    "health_record": true,
    "location_cache": true,
    "distance_unit": "km",
    "reconnecting_timeout": 15,
    "item_filter": {
      "1": { "keep" : 50 },
      "2": { "keep" : 50 },
      "3": { "keep" : 50 },
      "101": { "keep" : 10 },
      "102": { "keep" : 25 },
      "103": { "keep" : 25 },
      "201": { "keep" : 25 },
      "701": { "keep" : 100 }
    },
    "evolve_all": "false",
    "evolve_cp_min": 300,
    "evolve_captured": "NONE",
    "catch_randomize_reticle_factor": 1.0,
    "catch_randomize_spin_factor": 1.0,
    "catch": {
      "any": {"catch_above_cp": 0, "catch_above_iv": 0, "logic": "or" },

      "// Pokemons with example": { "always_catch": true },
      "// Gets filtered with release parameters": {},

      "// Legendary pokemons (Goes under S-Tier)": {},
      "Lapras": { "always_catch": true },
      "Moltres": { "always_catch": true },
      "Zapdos": { "always_catch": true },
      "Articuno": { "always_catch": true },

      "// S-Tier pokemons (if pokemon can be evolved into tier, list the representative)": {},
      "Mewtwo": { "always_catch": true },
      "Dragonite": { "always_catch": true },
      "Snorlax": { "always_catch": true },
      "// Mew evolves to Mewtwo": {},
      "Mew": { "always_catch": true },
      "Arcanine": { "always_catch": true },
      "Vaporeon": { "always_catch": true },
      "Gyarados": { "always_catch": true },
      "Exeggutor": { "always_catch": true },
      "Muk": { "always_catch": true },
      "Weezing": { "always_catch": true },
      "Flareon": { "always_catch": true },

      "// Growlithe evolves to Arcanine": {},
      "Growlithe": { "always_catch": true },
      "// Dragonair evolves to Dragonite": {},
      "Dragonair": { "always_catch": true },
      "// Grimer evolves to Muk": {},
      "Grimer": { "always_catch": true },

      "// Magikarp evolves to Gyarados": {},
      "Magikarp": { "always_catch": true },
      "// Exeggcute evolves to Exeggutor": {},
      "Exeggcute": { "always_catch": true },
      "// Eevee evolves to many versions, like Vaporeon, Flareon": {},
      "Eevee": { "always_catch": true },

      "// A-Tier pokemons": {},
      "Slowbro": { "always_catch": true },
      "Victreebel": { "always_catch": true },
      "Machamp": { "always_catch": true },
      "Poliwrath": { "always_catch": true },
      "Clefable": { "always_catch": true },
      "Nidoking": { "always_catch": true },
      "Venusaur": { "always_catch": true },
      "Charizard": { "always_catch": true },
      "Golduck": { "always_catch": true },
      "Nidoqueen": { "always_catch": true },
      "Vileplume": { "always_catch": true },
      "Blastoise": { "always_catch": true },
      "Omastar":  { "always_catch": true },
      "Aerodactyl": { "always_catch": true },
      "Golem": { "always_catch": true },
      "Wigglytuff": { "always_catch": true },
      "Dewgong": { "always_catch": true },
      "Ninetales": { "always_catch": true },
      "Magmar": { "always_catch": true },
      "Kabutops": { "always_catch": true },
      "Electabuzz": { "always_catch": true },
      "Starmie": { "always_catch": true },
      "Jolteon": { "always_catch": true },
      "Rapidash": { "always_catch": true },
      "Pinsir": { "always_catch": true },
      "Scyther": { "always_catch": true },
      "Tentacruel": { "always_catch": true },
      "Gengar": { "always_catch": true },
      "Hypno": { "always_catch": true },
      "Pidgeot": { "always_catch": true },
      "Rhydon": { "always_catch": true },
      "Seaking": { "always_catch": true },
      "Kangaskhan": { "always_catch": true }
    },
    "release": {
      "any": {"release_below_cp": 0, "release_below_iv": 0, "logic": "or" },

      "// Legendary pokemons (Goes under S-Tier)": {},
      "Lapras": { "release_below_cp": 1041, "release_below_iv": 0.8, "logic": "and" },
      "Moltres": { "release_below_cp": 1132, "release_below_iv": 0.8, "logic": "and" },
      "Zapdos": { "release_below_cp": 1087, "release_below_iv": 0.8, "logic": "and" },
      "Articuno": { "release_below_cp": 1039, "release_below_iv": 0.8, "logic": "and" },

      "// S-Tier pokemons (if pokemon can be evolved into tier, list the representative)": {},
      "Mewtwo": { "release_below_cp": 1447, "release_below_iv": 0.8, "logic": "and"},
      "Dragonite": { "release_below_cp": 1221, "release_below_iv": 0.8, "logic": "and" },
      "Snorlax": { "release_below_cp": 1087, "release_below_iv": 0.8, "logic": "and" },
      "// Mew evolves to Mewtwo": {},
      "Mew": { "release_below_cp": 1152, "release_below_iv": 0.8, "logic": "and" },
      "Arcanine": { "release_below_cp": 1041, "release_below_iv": 0.8, "logic": "and" },
      "Vaporeon": { "release_below_cp": 984, "release_below_iv": 0.8, "logic": "and" },
      "Gyarados": { "release_below_cp": 938, "release_below_iv": 0.8, "logic": "and" },
      "Exeggutor": { "release_below_cp": 1032, "release_below_iv": 0.8, "logic": "and" },
      "Muk": { "release_below_cp": 909, "release_below_iv": 0.8, "logic": "and" },
      "Weezing": { "release_below_cp": 784, "release_below_iv": 0.8, "logic": "and" },
      "Flareon": { "release_below_cp": 924, "release_below_iv": 0.8, "logic": "and" },

      "// Growlithe evolves to Arcanine": {},
      "Growlithe": { "release_below_cp": 465, "release_below_iv": 0.8, "logic": "and" },
      "// Dragonair evolves to Dragonite": {},
      "Dragonair": { "release_below_cp": 609, "release_below_iv": 0.8, "logic": "and" },
      "// Grimer evolves to Muk": {},
      "Grimer": { "release_below_cp": 448, "release_below_iv": 0.8, "logic": "and" },
      "// Magikarp evolves to Gyarados": {},
      "Magikarp": { "release_below_cp": 91, "release_below_iv": 0.8, "logic": "and" },
      "// Exeggcute evolves to Exeggutor": {},
      "Exeggcute": { "release_below_cp": 384, "release_below_iv": 0.8, "logic": "and" },
      "// Eevee evolves to many versions, like Vaporeon, Flareon": {},
      "Eevee": { "release_below_cp": 376, "release_below_iv": 0.8, "logic": "and" },

      "// A-Tier pokemons": {},
      "Slowbro": { "release_below_cp": 907, "release_below_iv": 0.8, "logic": "and" },
      "Victreebel": { "release_below_cp": 883, "release_below_iv": 0.8, "logic": "and" },
      "Machamp": { "release_below_cp": 907, "release_below_iv": 0.8, "logic": "and" },
      "Poliwrath": { "release_below_cp": 876, "release_below_iv": 0.8, "logic": "and" },
      "Clefable": { "release_below_cp": 837, "release_below_iv": 0.8, "logic": "and" },
      "Nidoking": { "release_below_cp": 864, "release_below_iv": 0.8, "logic": "and" },
      "Venusaur": { "release_below_cp": 902, "release_below_iv": 0.8, "logic": "and" },
      "Charizard": { "release_below_cp": 909, "release_below_iv": 0.8, "logic": "and" },
      "Golduck": { "release_below_cp": 832, "release_below_iv": 0.8, "logic": "and" },
      "Nidoqueen": { "release_below_cp": 868, "release_below_iv": 0.8, "logic": "and" },
      "Vileplume": { "release_below_cp": 871, "release_below_iv": 0.8, "logic": "and" },
      "Blastoise": { "release_below_cp": 888, "release_below_iv": 0.8, "logic": "and" },
      "Omastar":  { "release_below_cp": 780, "release_below_iv": 0.8, "logic": "and" },
      "Aerodactyl": { "release_below_cp": 756, "release_below_iv": 0.8, "logic": "and" },
      "Golem": { "release_below_cp": 804, "release_below_iv": 0.8, "logic": "and" },
      "Wigglytuff": { "release_below_cp": 760, "release_below_iv": 0.8, "logic": "and" },
      "Dewgong": { "release_below_cp": 748, "release_below_iv": 0.8, "logic": "and" },
      "Ninetales": { "release_below_cp": 763, "release_below_iv": 0.8, "logic": "and" },
      "Magmar": { "release_below_cp": 792, "release_below_iv": 0.8, "logic": "and" },
      "Kabutops": { "release_below_cp": 744, "release_below_iv": 0.8, "logic": "and" },
      "Electabuzz": { "release_below_cp": 739, "release_below_iv": 0.8, "logic": "and" },
      "Starmie": { "release_below_cp": 763, "release_below_iv": 0.8, "logic": "and" },
      "Jolteon": { "release_below_cp": 746, "release_below_iv": 0.8, "logic": "and" },
      "Rapidash": { "release_below_cp": 768, "release_below_iv": 0.8, "logic": "and" },
      "Pinsir": { "release_below_cp": 741, "release_below_iv": 0.8, "logic": "and" },
      "Scyther": { "release_below_cp": 724, "release_below_iv": 0.8, "logic": "and" },
      "Tentacruel": { "release_below_cp": 775, "release_below_iv": 0.8, "logic": "and" },
      "Gengar": { "release_below_cp": 724, "release_below_iv": 0.8, "logic": "and" },
      "Hypno": { "release_below_cp": 763, "release_below_iv": 0.8, "logic": "and" },
      "Pidgeot": { "release_below_cp": 729, "release_below_iv": 0.8, "logic": "and" },
      "Rhydon": { "release_below_cp": 782, "release_below_iv": 0.8, "logic": "and" },
      "Seaking": { "release_below_cp": 712, "release_below_iv": 0.8, "logic": "and" },
      "Kangaskhan": { "release_below_cp": 712, "release_below_iv": 0.8, "logic": "and" },

      "// Koffing evolves to Weezing (A-Tier)": {},
      "Koffing": { "release_below_cp": 403, "release_below_iv": 0.8, "logic": "and" },

      "// Below is B-tier and lower pokemons": {},
      "Caterpie": { "release_below_cp": 156, "release_below_iv": 0.8, "logic": "and" },
      "Weedle": { "release_below_cp": 156, "release_below_iv": 0.8, "logic": "and" },
      "Diglett": { "release_below_cp": 158, "release_below_iv": 0.8, "logic": "and" },
      "Metapod": { "release_below_cp": 168, "release_below_iv": 0.8, "logic": "and" },
      "Kakuna": { "release_below_cp": 170, "release_below_iv": 0.8, "logic": "and" },
      "Rattata": { "release_below_cp": 204, "release_below_iv": 0.8, "logic": "and" },
      "Abra": { "release_below_cp": 208, "release_below_iv": 0.8, "logic": "and" },
      "Zubat": { "release_below_cp": 225, "release_below_iv": 0.8, "logic": "and" },
      "Chansey": { "release_below_cp": 235, "release_below_iv": 0.8, "logic": "and" },
      "Pidgey": { "release_below_cp": 237, "release_below_iv": 0.8, "logic": "and" },
      "Spearow": { "release_below_cp": 240, "release_below_iv": 0.8, "logic": "and" },
      "Meowth": { "release_below_cp": 264, "release_below_iv": 0.8, "logic": "and" },
      "Krabby": { "release_below_cp": 276, "release_below_iv": 0.8, "logic": "and" },
      "Sandshrew": { "release_below_cp": 278, "release_below_iv": 0.8, "logic": "and" },
      "Poliwag": { "release_below_cp": 278, "release_below_iv": 0.8, "logic": "and" },
      "Horsea": { "release_below_cp": 278, "release_below_iv": 0.8, "logic": "and" },
      "Gastly": { "release_below_cp": 280, "release_below_iv": 0.8, "logic": "and" },
      "Ekans": { "release_below_cp": 288, "release_below_iv": 0.8, "logic": "and" },
      "Shellder": { "release_below_cp": 288, "release_below_iv": 0.8, "logic": "and" },
      "Vulpix": { "release_below_cp": 290, "release_below_iv": 0.8, "logic": "and" },
      "Voltorb": { "release_below_cp": 292, "release_below_iv": 0.8, "logic": "and" },
      "Geodude": { "release_below_cp": 297, "release_below_iv": 0.8, "logic": "and" },
      "Doduo": { "release_below_cp": 297, "release_below_iv": 0.8, "logic": "and" },
      "Onix": { "release_below_cp": 300, "release_below_iv": 0.8, "logic": "and" },
      "Mankey": { "release_below_cp": 307, "release_below_iv": 0.8, "logic": "and" },
      "Pikachu": { "release_below_cp": 309, "release_below_iv": 0.8, "logic": "and" },
      "Magnemite": { "release_below_cp": 312, "release_below_iv": 0.8, "logic": "and" },
      "Tentacool": { "release_below_cp": 316, "release_below_iv": 0.8, "logic": "and" },
      "Paras": { "release_below_cp": 319, "release_below_iv": 0.8, "logic": "and" },
      "Jigglypuff": { "release_below_cp": 321, "release_below_iv": 0.8, "logic": "and" },
      "Ditto": { "release_below_cp": 321, "release_below_iv": 0.8, "logic": "and" },
      "Staryu": { "release_below_cp": 326, "release_below_iv": 0.8, "logic": "and" },
      "Charmander": { "release_below_cp": 333, "release_below_iv": 0.8, "logic": "and" },
      "Goldeen": { "release_below_cp": 336, "release_below_iv": 0.8, "logic": "and" },
      "Squirtle": { "release_below_cp": 352, "release_below_iv": 0.8, "logic": "and" },
      "Cubone": { "release_below_cp": 352, "release_below_iv": 0.8, "logic": "and" },
      "Venonat": { "release_below_cp": 360, "release_below_iv": 0.8, "logic": "and" },
      "Bulbasaur": { "release_below_cp": 374, "release_below_iv": 0.8, "logic": "and" },
      "Drowzee": { "release_below_cp": 374, "release_below_iv": 0.8, "logic": "and" },
      "Machop": { "release_below_cp": 381, "release_below_iv": 0.8, "logic": "and" },
      "Psyduck": { "release_below_cp": 386, "release_below_iv": 0.8, "logic": "and" },
      "Seel": { "release_below_cp": 386, "release_below_iv": 0.8, "logic": "and" },
      "Kabuto": { "release_below_cp": 386, "release_below_iv": 0.8, "logic": "and" },
      "Bellsprout": { "release_below_cp": 391, "release_below_iv": 0.8, "logic": "and" },
      "Omanyte": { "release_below_cp": 391, "release_below_iv": 0.8, "logic": "and" },
      "Kadabra": { "release_below_cp": 396, "release_below_iv": 0.8, "logic": "and" },
      "Oddish": { "release_below_cp": 400, "release_below_iv": 0.8, "logic": "and" },
      "Dugtrio": { "release_below_cp": 408, "release_below_iv": 0.8, "logic": "and" },
      "Rhyhorn": { "release_below_cp": 412, "release_below_iv": 0.8, "logic": "and" },
      "Clefairy": { "release_below_cp": 420, "release_below_iv": 0.8, "logic": "and" },
      "Slowpoke": { "release_below_cp": 424, "release_below_iv": 0.8, "logic": "and" },
      "Pidgeotto": { "release_below_cp": 427, "release_below_iv": 0.8, "logic": "and" },
      "Farfetch'd": { "release_below_cp": 441, "release_below_iv": 0.8, "logic": "and" },
      "Poliwhirl": { "release_below_cp": 468, "release_below_iv": 0.8, "logic": "and" },
      "Nidorino": { "release_below_cp": 480, "release_below_iv": 0.8, "logic": "and" },
      "Haunter": { "release_below_cp": 482, "release_below_iv": 0.8, "logic": "and" },
      "Nidorina": { "release_below_cp": 489, "release_below_iv": 0.8, "logic": "and" },
      "Graveler": { "release_below_cp": 501, "release_below_iv": 0.8, "logic": "and" },
      "Beedrill": { "release_below_cp": 504, "release_below_iv": 0.8, "logic": "and" },
      "Raticate": { "release_below_cp": 504, "release_below_iv": 0.8, "logic": "and" },
      "Butterfree": { "release_below_cp": 508, "release_below_iv": 0.8, "logic": "and" },
      "Hitmonlee": { "release_below_cp": 520, "release_below_iv": 0.8, "logic": "and" },
      "Ponyta": { "release_below_cp": 530, "release_below_iv": 0.8, "logic": "and" },
      "Hitmonchan": { "release_below_cp": 530, "release_below_iv": 0.8, "logic": "and" },
      "Charmeleon": { "release_below_cp": 544, "release_below_iv": 0.8, "logic": "and" },
      "Wartortle": { "release_below_cp": 552, "release_below_iv": 0.8, "logic": "and" },
      "Persian": { "release_below_cp": 568, "release_below_iv": 0.8, "logic": "and" },
      "Lickitung": { "release_below_cp": 568, "release_below_iv": 0.8, "logic": "and" },
      "Ivysaur": { "release_below_cp": 571, "release_below_iv": 0.8, "logic": "and" },
      "Electrode": { "release_below_cp": 576, "release_below_iv": 0.8, "logic": "and" },
      "Marowak": { "release_below_cp": 578, "release_below_iv": 0.8, "logic": "and" },
      "Gloom": { "release_below_cp": 590, "release_below_iv": 0.8, "logic": "and" },
      "Porygon": { "release_below_cp": 590, "release_below_iv": 0.8, "logic": "and" },
      "Seadra": { "release_below_cp": 597, "release_below_iv": 0.8, "logic": "and" },
      "Jynx": { "release_below_cp": 600, "release_below_iv": 0.8, "logic": "and" },
      "Weepinbell": { "release_below_cp": 602, "release_below_iv": 0.8, "logic": "and" },
      "Tangela": { "release_below_cp": 607, "release_below_iv": 0.8, "logic": "and" },
      "Fearow": { "release_below_cp": 609, "release_below_iv": 0.8, "logic": "and" },
      "Parasect": { "release_below_cp": 609, "release_below_iv": 0.8, "logic": "and" },
      "Machoke": { "release_below_cp": 614, "release_below_iv": 0.8, "logic": "and" },
      "Arbok": { "release_below_cp": 616, "release_below_iv": 0.8, "logic": "and" },
      "Sandslash": { "release_below_cp": 631, "release_below_iv": 0.8, "logic": "and" },
      "Alakazam": { "release_below_cp": 633, "release_below_iv": 0.8, "logic": "and" },
      "Kingler": { "release_below_cp": 636, "release_below_iv": 0.8, "logic": "and" },
      "Dodrio": { "release_below_cp": 640, "release_below_iv": 0.8, "logic": "and" },
      "Tauros": { "release_below_cp": 643, "release_below_iv": 0.8, "logic": "and" },
      "Primeape": { "release_below_cp": 650, "release_below_iv": 0.8, "logic": "and" },
      "Magneton": { "release_below_cp": 657, "release_below_iv": 0.8, "logic": "and" },
      "Venomoth": { "release_below_cp": 660, "release_below_iv": 0.8, "logic": "and" },
      "Golbat": { "release_below_cp": 672, "release_below_iv": 0.8, "logic": "and" },
      "Raichu": { "release_below_cp": 708, "release_below_iv": 0.8, "logic": "and" },
      "Cloyster": { "release_below_cp": 717, "release_below_iv": 0.8, "logic": "and"},
      "Mr. Mime": { "release_below_cp": 650, "release_below_iv": 0.8, "logic": "and" }
    },
    "vips" : {
         "Any pokemon put here directly force to use Berry & Best Ball to capture, to secure the capture rate!": {},
        "any": {"catch_above_cp": 1200, "catch_above_iv": 0.9, "logic": "or" },
        "Lapras": {},
        "Moltres": {},
        "Zapdos": {},
        "Articuno": {},

        "// S-Tier pokemons (if pokemon can be evolved into tier, list the representative)": {},
        "Mewtwo": {},
        "Dragonite": {},
        "Snorlax": {},
        "// Mew evolves to Mewtwo": {},
        "Mew": {},
        "Arcanine": {},
        "Vaporeon": {},
        "Gyarados": {},
        "Exeggutor": {},
        "Muk": {},
        "Weezing": {},
        "Flareon": {}

    }
}`

**Comment by MFizz** [comment]
Ok, added a new flag: lured to search for lured clusters. As new stops are being lured and old expire, destiniy will change over time. If nothing changes bot will stay in the middle of reachable (as set in radius - perhaps more than 50m?) lured pokestops. With no lured pokestops it will search for the biggest non-lured cluster instead.

File: pokemongo_bot/test/follow_cluster_test.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import unittest, pickle, os
```
Comment (TheSavior) [inline pokemongo_bot/test/follow_cluster_test.py]: Oh man, I'm so excited by this file. Thank you so so much for writing tests

**Comment by douglascamata** [comment]
@MFizz merge conflicts again, sorry... can you solve, please? :/

**Comment by douglascamata** [comment]
👍

**Comment by TheSavior** [comment]
@MFizz, your travis build failed

**Comment by DayBr3ak** [comment]
You must have done something wrong with your branch you have .... 19 files changed and they are from other PR

File: requirements.txt [hunk 5]
```diff
@@ -1,3 +1,6 @@
+mock==1.3.0
```
Comment (DayBr3ak) [inline requirements.txt]: there is already mock==2.0 in the dependencies

File: pokemongo_bot/test/follow_cluster_test.py (hunk 4 above)
Comment (DayBr3ak) [inline pokemongo_bot/test/follow_cluster_test.py]: This file should be in {root_folder}/tests/ directory

**Comment by TheSavior** [comment]
:+1:

**Comment by douglascamata** [comment]
👍
